hack/libe2e.sh

```diff
@@ -101,3 +99,49 @@ function stdout_gt() {
     fi
     return 1
 }
+
+function dump_debug_logs() {
+    local namespace="${1}"
+    if [ "${namespace}" != "" ]; then
+        local output_dir="$(pwd)/_artifacts/${namespace}/dump.txt"
+        echo "Dumping cluster state to ${output_dir}"
+        mkdir -p "${output_dir}"
+        kubectl cluster-info dump --namespaces "${namespace}" > "${output_dir}/dump.txt" || true
+    fi
+}
+
+function ignore_expected_controller_errors() {
+    # Ignore the following error types:
+    # E1103 14:58:06.819858       1 reflector.go:205] github.com/jetstack/navigator/pkg/client/informers/externalversions/factory.go:68: Failed to list *v1alpha1.Pilot: the server could not find the requested resource (get pilots.navigator.jetstack.io)
+    # E1108 14:18:37.610718       1 reflector.go:205] github.com/jetstack/navigator/pkg/client/informers/externalversions/factory.go:68: Failed to list *v1alpha1.Pilot: an error on the server ("Error: 'dial tcp 10.0.0.233:443: getsockopt: connection refused'\nTrying to reach: 'https://10.0.0.233:443/apis/navigator.jetstack.io/v1alpha1/pilots?resourceVersion=0'") has prevented the request from succeeding (get pilots.navigator.jetstack.io)
+    egrep --invert-match \
+          -e 'Failed to list \*v1alpha1\.\w+:\s+the server could not find the requested resource\s+\(get \w+\.navigator\.jetstack\.io\)$' \
+          -e 'Failed to list \*v1alpha1\.\w+:\s+an error on the server \([^)]+\) has prevented the request from succeeding\s+\(get \w+\.navigator\.jetstack\.io\)$'
+}
+
+function test_logged_errors() {
+    local namespace="${1}"
+    if kubectl logs -c controller -l app=navigator,component=controller \
+            | egrep '^E[0-9]{4} ' \
+            | ignore_expected_controller_errors
+    then
+        fail_test "Unexpected errors in controller logs" "${namespace}"
+    fi
+    if kubectl logs -c apiserver -l app=navigator,component=apiserver \
+            | egrep '^E[0-9]{4} '
+    then
+        fail_test "Unexpected errors in apiserver logs" "${namespace}"
+    fi
+}
```

Comment:
Ok, you win :-) Let's just delete this test now that we have full logs for each test run.